src/globals.cpp

```diff
@@ -43,5 +43,22 @@ Real receive_boundary_buffer_timeout;
 // timeouts for tasks
 Real current_task_runtime_sec;
 
+namespace cell_centered_refinement {
+// If the info object has more buffers than this, do
+// hierarchical parallelism. If it does not, loop over buffers on the
+// host and launch kernels manually.
+//
+// min_num_bufs = 1 implies that the old per-buffer machinery doesn't
+// use hierarchical parallelism. This also means that for
+// prolongation/restriction over a whole meshblock, hierarchical
+// parallelism is not used, which is probably important for
+// re-meshing.
```

Comment:
Isn't hierarchical parallelism used if `num_bufs > min_num_bufs` and otherwise multiple kernels are launched?

Reply:
yes that's right. I think that's what I wrote here. Maybe this verbose comment is too much. I'll simplify the statement to be a bit more clear.